src/batch/src/Job/Item/AbstractElementDecorator.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yokai\Batch\Job\Item;
+
+use Yokai\Batch\Job\JobExecutionAwareInterface;
+use Yokai\Batch\Job\JobExecutionAwareTrait;
+use Yokai\Batch\Job\JobParametersAwareInterface;
+use Yokai\Batch\Job\SummaryAwareInterface;
+
+/**
+ * Extends this class if you want to create a decorator of :
+ * - {@see ItemReaderInterface}
+ * - {@see ItemProcessorInterface}
+ * - {@see ItemWriterInterface}
+ *
+ * This class covers interfaces that the decorated instance might have :
+ * - {@see InitializableInterface}
+ * - {@see FlushableInterface}
+ * - {@see JobExecutionAwareInterface}
+ * - {@see SummaryAwareInterface}
+ * - {@see JobParametersAwareInterface}
+ */
+abstract class AbstractElementDecorator implements
+    InitializableInterface,
+    FlushableInterface,
+    JobExecutionAwareInterface
+{
+    use ElementConfiguratorTrait;
```

Comment:
Do you still need those traits now that there is an abstract class taking care of the logic? Couldn't you replace them by private methods?

Reply:
Still required, this trait is also used in several classes cannot be child of this abstract class : 

- https://github.com/yokai-php/batch-src/blob/0.x/src/batch/src/Job/Item/ItemJob.php#L14
- https://github.com/yokai-php/batch-src/blob/0.x/src/batch/src/Job/Item/Processor/RoutingProcessor.php#L22
- https://github.com/yokai-php/batch-src/blob/0.x/src/batch/src/Job/Item/Writer/RoutingWriter.php#L22